src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -69,6 +81,24 @@ cdef class RPCState:
         if self.server._status == AIO_SERVER_STATUS_STOPPED:
             raise _ServerStoppedError(_SERVER_STOPPED_DETAILS)
 
+    cdef int get_write_flag(self):
+        if self.disable_next_compression:
+            self.disable_next_compression = False
+            return WriteFlag.no_compress
+        else:
+            return _EMPTY_FLAG
+
+    cdef Operation create_send_initial_metadata_op_if_not_sent(self):
+        if self.metadata_sent:
+            return None
+
+        cdef SendInitialMetadataOperation op = SendInitialMetadataOperation(
+            _augment_metadata(_IMMUTABLE_EMPTY_METADATA, self.compression_algorithm),
+            _EMPTY_FLAG
+        )
+        self.metadata_sent = True
```

Comment:
This is a bit odd. The naming implies that this is a pure function, but it's actually side-effecting. And this is actually a bit disingenuous; the metadata hasn't actually been sent yet.